azure-kusto-ingest/azure/kusto/ingest/descriptors.py

```diff
@@ -171,14 +173,20 @@ def upload_from_different_descriptor(
         :return new BlobDescriptor instance
         """
         blob_name = "{db}__{table}__{guid}__{file}".format(db=database, table=table, guid=descriptor.source_id, file=descriptor.stream_name)
-        random_container = random.choice(containers)
-        try:
-            blob_service = BlobServiceClient(random_container.account_uri, proxies=proxy_dict)
-            blob_client = blob_service.get_blob_client(container=random_container.object_name, blob=blob_name)
-            blob_client.upload_blob(data=stream, timeout=timeout)
-        except Exception as e:
-            raise KustoBlobError(e)
-        return BlobDescriptor(blob_client.url, descriptor.size, descriptor.source_id)
+
+        retries_left = min(max_retries, len(containers))
+        for container in containers:
+            try:
+                blob_service = BlobServiceClient(container.account_uri, proxies=proxy_dict)
+                blob_client = blob_service.get_blob_client(container=container.object_name, blob=blob_name)
+                blob_client.upload_blob(data=stream, timeout=timeout)
+                resource_manager.Report_resource_usage_result(container.storage_account_name, True)
```

Comment:
I don't like this dependency, having resource manager here.
This whole function doesn't really belong here, really, since it's static and is only called from inside IngestClient anyway.

Make it a method inside IngestClient, which will save some awkwardness.

Reply:
Yochai wanted to send the usage data to resource_manager , therefore it is a method of resource_manager

Reply:
I meant the method "upload_from_different_descriptor".
Move the entire method to ingest_client, rename it "upload_blob" or something similar.